runtime-light/stdlib/crypto/crypto-functions.cpp

```diff
@@ -343,3 +347,97 @@ task_t<Optional<string>> f$openssl_decrypt(string data, const string &method, co
   }
   co_return string{response.inner.value.data(), static_cast<string::size_type>(response.inner.value.size())};
 }
+
+namespace {
+constexpr std::pair<const char *, tl::HashAlgorithm> HASH_ALGOS[] = {{"md5", tl::HashAlgorithm::MD5},       {"sha1", tl::HashAlgorithm::SHA1},
+                                                                     {"sha224", tl::HashAlgorithm::SHA224}, {"sha256", tl::HashAlgorithm::SHA256},
+                                                                     {"sha384", tl::HashAlgorithm::SHA384}, {"sha512", tl::HashAlgorithm::SHA512}};
+
+std::optional<tl::HashAlgorithm> parse_hash_algorithm(const string &algo_str) noexcept {
```

Comment:
It looks redundant... Do you really need such a complexity to just check that an array contains a string?

Reply:
What exactly do you consider redundant?
`find_if` instead of for loop?

Reply:
I'm just curious whether we can simplify it a bit

Reply:
Idk how, as for me this code is simple enough. It can be rewritten with additional 2 for loops, but it wouldn't become simpler.